builtins: implement regexp_split_to{table,array}

otan opened the pull request:
Refs: #51818

Release note (sql change): Implement the regexp_split_to_table and
regexp_split_to_array builtins.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/52479)
<!-- Reviewable:end -->

rohany commented on pkg/sql/logictest/testdata/logic_test/builtin_function:
```diff
@@ -2786,9 +2790,32 @@ SELECT (SELECT * FROM [SHOW crdb_version]) = version()
+subtest regexp_split
+
+query T
+SELECT regexp_split_to_table('3a0A1', 'a')
```
can you test with some regexes

otan replied:
Done!

rohany commented on pkg/sql/sem/builtins/builtins.go:
```diff
@@ -1437,26 +1458,56 @@ var builtins = map[string]builtinDefinition{
+			},
+			ReturnType: tree.FixedReturnType(types.MakeArray(types.String)),
+			Fn: func(ctx *tree.EvalContext, args tree.Datums) (tree.Datum, error) {
+				words, err := regexpSplit(ctx, args, false /* hasFlags */)
```
can probably make this function body a short helper

otan replied:
Done!

otan committed:
builtins: implement regexp_split_to{table,array}

Release note (sql change): Implement the regexp_split_to_table and
regexp_split_to_array builtins.

otan commented:
bors r=rohany
